engine/src/main/java/org/terasology/engine/world/chunks/pipeline/ChunkProcessingPipeline.java

```diff
@@ -53,10 +53,19 @@
      * Create ChunkProcessingPipeline.
      */
     public ChunkProcessingPipeline(Function<Vector3ic, Chunk> chunkProvider, Flux<Chunk> chunkStream) {
+        this(chunkProvider, chunkStream, true);
+    }
+
+    /**
+     * @param threaded Whether to use worker threads to generate chunks. If it's false, all processing will be done on the main thread.
+     * It should be set to false in tests that request processing of specific chunks.
+     * Always true in a real game.
```

Comment:
I think that doing away with the complexity of multi-threading during tests hopefully helps to fight flakiness and instability. 

However, this also means that we're not testing the "real thing" here. The good part still is that if something goes wrong in the end at runtime we know to start looking at the concurrency parts...

@keturn might this pitfall be also occurring in Pathfinding, even without using Reactor?

Reply:
I do agree, it's unfortunate that the concurrent version doesn't work reliably on CI. I'd kind of still like to be able to run it concurrently on my computer, since it does work reliably on a normal computer, but I don't know if there's a nice way to do that.

Reply:
A _lot_ of the complexity of this code is there explicitly for the reason of allowing chunk generation to happen in parallel.

If there were only a single-threaded sink processing things from a queue, there's a lot of synchronization and `currentyProcessing` sorts of variables we wouldn't need.

If we are never testing the multi-threaded scenarios, we're seriously undermining the value of our tests.